src/frontend/src/elements/YearSwitcher.tsx

```diff
@@ -0,0 +1,39 @@
+import { useSearchParams } from "react-router-dom";
+
+
+function getCurrentYear() {
+    return new Date().getFullYear();
+}
+
+const LAST = -1;
+
+
+function useYearManipulation(): [number, (y: number) => void] {
+    const [searchParams, setSearchParams] = useSearchParams();
+
+    function setYear(year: number) {
+        setSearchParams((prev) => {
+            prev.set("year", `${year}`);
+            return prev;
+        });
+    }
+
+    const years = searchParams.getAll("year");
+    const year = years.length ? parseInt(years.at(LAST)!) : getCurrentYear();
+
+    return [year, setYear];
+}
+
+const ONE = 1;
+const STARTING_YEAR = 2000;
+
+
+export default function YearSwitcher() {
```

Comment:
Wahrscheinlich wird es praktischer, den year state über die props mitzugeben (also: year, hasPrevYear, hasNextYear, incrementYear, decrementYear). Ich glaube der State muss in diesem Fall einfach in der Page liegen, da sich ja der Graf anhand von dem State aktuallisiern muss

Reply:
nope, macht das ganze komplizierter